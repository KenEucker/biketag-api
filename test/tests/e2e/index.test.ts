import {
  getGameAsync,
  getCurrentTagAsync,
  get10TagsAsync,
  getTag1Async,
  bikeTagSanityInstance,
  bikeTagImgurInstance,
  bikeTagRedditInstance,
  bikeTagTwitterInstance,
} from '../../../examples/node'

test('BikeTagClient integration tests run when they can', async () => {
  expect(true).toBeTruthy()
})

if (bikeTagSanityInstance) {
  test('BikeTagClient sanity instance can retrieve game data', async () => {
    const testGameData = await getGameAsync('test', bikeTagSanityInstance)

    expect(testGameData).toMatchSnapshot()
  })

  test('BikeTagClient sanity instance can retrieve tag 1 data', async () => {
    const testTagData = await getTag1Async('test', bikeTagSanityInstance)

    expect(testTagData).toMatchSnapshot()
  })

  test('BikeTagClient sanity instance can retrieve latest tag data', async () => {
    const testTagData = await getCurrentTagAsync('test', bikeTagSanityInstance)

    expect(testTagData).toMatchSnapshot()
  })

  test('BikeTagClient sanity instance can retrieve all tags data', async () => {
    const testTagsData = await get10TagsAsync('test', bikeTagSanityInstance)

    expect(testTagsData).toMatchSnapshot()
  })
}

if (bikeTagImgurInstance) {
  test('BikeTagClient imgur instance can retrieve tag 1 data', async () => {
    const testTagData = await getTag1Async('test', bikeTagImgurInstance)

    expect(testTagData).toMatchSnapshot()
  })

  test('BikeTagClient imgur instance can retrieve latest tag data', async () => {
    const testTagData = await getCurrentTagAsync('test', bikeTagImgurInstance)

    expect(testTagData).toMatchSnapshot()
  })

  test('BikeTagClient imgur instance can retrieve all tags data', async () => {
    const testTagsData = await get10TagsAsync('test', bikeTagImgurInstance)

    expect(testTagsData).toMatchSnapshot()
  })
}

if (bikeTagRedditInstance) {
  test('BikeTagClient reddit instance can retrieve tag 1 data', async () => {
    const testTagData = await getTag1Async(
      'test',
      bikeTagRedditInstance,
      false,
      {
        source: 'reddit',
      }
    )

    expect(testTagData).toMatchSnapshot()
  })

  test('BikeTagClient reddit instance can retrieve latest tag data', async () => {
    const testTagData = await getCurrentTagAsync(
      'test',
      bikeTagRedditInstance,
      false,
      {
        source: 'reddit',
      }
    )

    expect(testTagData).toMatchSnapshot()
  })

  /// Test takes too long
  // test('BikeTagClient reddit instance can retrieve all tags data', async () => {
  //   const testTagsData = await get10TagsAsync('test', bikeTagRedditInstance, false, { source: 'reddit', limit: 100 })

  //   expect(testTagsData).toMatchSnapshot()
  // })
}

if (bikeTagTwitterInstance) {
  test('BikeTagClient twitter instance can retrieve tag 1 data', async () => {
    const testTagData = await getTag1Async('test', bikeTagTwitterInstance)

    expect(testTagData).toMatchSnapshot()
  })

  test('BikeTagClient twitter instance can retrieve latest tag data', async () => {
    const testTagData = await getCurrentTagAsync('test', bikeTagTwitterInstance)

    expect(testTagData).toMatchSnapshot()
  })

  test('BikeTagClient twitter instance can retrieve all tags data', async () => {
    const testTagsData = await get10TagsAsync('test', bikeTagTwitterInstance)

    expect(testTagsData).toMatchSnapshot()
  })
}
