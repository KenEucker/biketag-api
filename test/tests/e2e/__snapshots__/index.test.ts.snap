// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BikeTagClient imgur instance can retrieve all tags data 1`] = `
Object {
  "data": Array [
    Object {
      "discussionUrl": "https://redd.it/rnv8j5",
      "foundImageUrl": undefined,
      "foundLocation": undefined,
      "foundPlayer": null,
      "foundTime": undefined,
      "game": "portland",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "From one Bigfoot to another… actually just past the another",
      "mysteryImageUrl": "https://i.imgur.com/AIPoZlX.jpg",
      "mysteryPlayer": "Yossarian",
      "mysteryTime": 1640380734,
      "name": "portland-tag-534",
      "playerId": undefined,
      "slug": "portland-tag-534",
      "tagnumber": 534,
    },
    Object {
      "discussionUrl": "https://redd.it/rmg4f8",
      "foundImageUrl": "https://i.imgur.com/Avdyzof.jpg",
      "foundLocation": "13th andlambert",
      "foundPlayer": "Yossarian",
      "foundTime": 1640380712,
      "game": "portland",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "Outside apartments that share a name with a Hunger Games main character",
      "mysteryImageUrl": "https://i.imgur.com/dMuPFty.jpg",
      "mysteryPlayer": "pueo-pace",
      "mysteryTime": 1640210734,
      "name": "portland-tag-533",
      "playerId": undefined,
      "slug": "portland-tag-533",
      "tagnumber": 533,
    },
    Object {
      "discussionUrl": "https://redd.it/risne9",
      "foundImageUrl": "https://i.imgur.com/GL4yscw.jpg",
      "foundLocation": "Trolly Trail, near the  SE River Road and SE McLoughlin intersection",
      "foundPlayer": "pueo-pace",
      "foundTime": 1640210730,
      "game": "portland",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "rails, rails, rails to trails to trails end",
      "mysteryImageUrl": "https://i.imgur.com/vspDcMo.jpg",
      "mysteryPlayer": "The dynamic duo",
      "mysteryTime": 1639779933,
      "name": "portland-tag-532",
      "playerId": undefined,
      "slug": "portland-tag-532",
      "tagnumber": 532,
    },
    Object {
      "discussionUrl": undefined,
      "foundImageUrl": "https://i.imgur.com/hAhZ2sv.jpg",
      "foundLocation": "SE Milwaukie near Bybee Blvd.",
      "foundPlayer": "The dynamic duo",
      "foundTime": 1639779930,
      "game": "portland",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "It’s time to check out the Moreland n’hood",
      "mysteryImageUrl": "https://i.imgur.com/MBlpF4t.jpg",
      "mysteryPlayer": "Andrewkpdx",
      "mysteryTime": 1639538397,
      "name": "portland-tag-531",
      "playerId": undefined,
      "slug": "portland-tag-531",
      "tagnumber": 531,
    },
    Object {
      "discussionUrl": "https://redd.it/rgi5d1",
      "foundImageUrl": "https://i.imgur.com/HFSwwMj.jpg",
      "foundLocation": "Just off of Gladstone and SE 22nd",
      "foundPlayer": "Andrewkpdx",
      "foundTime": 1639538360,
      "game": "portland",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "a lone christmas tree out in the Bush",
      "mysteryImageUrl": "https://i.imgur.com/INAGM1z.jpg",
      "mysteryPlayer": "The dynamic duo",
      "mysteryTime": 1639518231,
      "name": "portland-tag-530",
      "playerId": undefined,
      "slug": "portland-tag-530",
      "tagnumber": 530,
    },
    Object {
      "discussionUrl": "https://redd.it/rai9kz",
      "foundImageUrl": "https://i.imgur.com/BHjxBEh.jpg",
      "foundLocation": "SE 53rd & Morrison",
      "foundPlayer": "The dynamic duo",
      "foundTime": 1639518228,
      "game": "portland",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "If you are heading north on a popular east side greenway take a right on a street named after a bridge or is the bridge named after the street?",
      "mysteryImageUrl": "https://i.imgur.com/4XAM4jt.jpg",
      "mysteryPlayer": "BlueThunder",
      "mysteryTime": 1638825763,
      "name": "portland-tag-529",
      "playerId": undefined,
      "slug": "portland-tag-529",
      "tagnumber": 529,
    },
    Object {
      "discussionUrl": "https://redd.it/r9ukt8",
      "foundImageUrl": "https://i.imgur.com/dqFEbo4.jpg",
      "foundLocation": "Grant park. I think all 3 statues have been played now",
      "foundPlayer": "BlueThunder",
      "foundTime": 1638825760,
      "game": "portland",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "This homage to a Portland children's author is just a few blocks south of the street she made famous",
      "mysteryImageUrl": "https://i.imgur.com/KpFhE2D.jpg?1",
      "mysteryPlayer": "The dynamic duo",
      "mysteryTime": 1638746043,
      "name": "portland-tag-528",
      "playerId": undefined,
      "slug": "portland-tag-528",
      "tagnumber": 528,
    },
    Object {
      "discussionUrl": "https://redd.it/r966bk",
      "foundImageUrl": "https://i.imgur.com/RdUHXg2.jpg",
      "foundLocation": "Skidmore Fountain",
      "foundPlayer": "The dynamic duo",
      "foundTime": 1638746040,
      "game": "portland",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "Granite and bronze and a circle of lions",
      "mysteryImageUrl": "https://i.imgur.com/5uWJ6gd.jpg",
      "mysteryPlayer": "pueo-pace",
      "mysteryTime": 1638673926,
      "name": "portland-tag-527",
      "playerId": undefined,
      "slug": "portland-tag-527",
      "tagnumber": 527,
    },
    Object {
      "discussionUrl": "https://redd.it/r5qs7l",
      "foundImageUrl": "https://i.imgur.com/rnOkIsx.jpg",
      "foundLocation": "NW Hoyt and NW 11th Ave",
      "foundPlayer": "pueo-pace",
      "foundTime": 1638673923,
      "game": "portland",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "Where Dianne gazes down upon the owl",
      "mysteryImageUrl": "https://i.imgur.com/fnjPAe1.jpg",
      "mysteryPlayer": "u/livehorribly",
      "mysteryTime": 1638288933,
      "name": "portland-tag-526",
      "playerId": undefined,
      "slug": "portland-tag-526",
      "tagnumber": 526,
    },
    Object {
      "discussionUrl": "https://redd.it/r3s21j",
      "foundImageUrl": "https://i.imgur.com/FHUJeqL.jpg",
      "foundLocation": "Peninsula park",
      "foundPlayer": "u/livehorribly",
      "foundTime": 1638288931,
      "game": "portland",
      "gps": Object {
        "alt": 0,
        "lat": 0,
        "long": 0,
      },
      "hint": "Stop to smell the roses",
      "mysteryImageUrl": "https://i.imgur.com/9P4Q6tH.jpg",
      "mysteryPlayer": "pueo-pace",
      "mysteryTime": 1638061586,
      "name": "portland-tag-525",
      "playerId": undefined,
      "slug": "portland-tag-525",
      "tagnumber": 525,
    },
  ],
  "source": "imgur",
  "status": 200,
  "success": true,
}
`;

exports[`BikeTagClient imgur instance can retrieve latest tag data 1`] = `
Object {
  "data": Object {
    "discussionUrl": "https://redd.it/rnv8j5",
    "foundImageUrl": undefined,
    "foundLocation": undefined,
    "foundPlayer": null,
    "foundTime": undefined,
    "game": "portland",
    "gps": Object {
      "alt": 0,
      "lat": 0,
      "long": 0,
    },
    "hint": "From one Bigfoot to another… actually just past the another",
    "mysteryImageUrl": "https://i.imgur.com/AIPoZlX.jpg",
    "mysteryPlayer": "Yossarian",
    "mysteryTime": 1640380734,
    "name": "portland-tag-534",
    "playerId": undefined,
    "slug": "portland-tag-534",
    "tagnumber": 534,
  },
  "source": "imgur",
  "status": 200,
  "success": true,
}
`;

exports[`BikeTagClient imgur instance can retrieve tag 1 data 1`] = `
Array [
  Object {
    "discussionUrl": undefined,
    "foundImageUrl": "https://i.imgur.com/buNHm8R.jpg",
    "foundLocation": undefined,
    "foundPlayer": "/u/youhavemyaxe",
    "foundTime": 1527392008,
    "game": "portland",
    "gps": Object {
      "alt": 0,
      "lat": 0,
      "long": 0,
    },
    "hint": null,
    "mysteryImageUrl": "https://i.imgur.com/KOAUXhp.jpg",
    "mysteryPlayer": "/u/TundraWolf_",
    "mysteryTime": 1527391986,
    "name": "portland-tag-1",
    "playerId": undefined,
    "slug": "portland-tag-1",
    "tagnumber": 1,
  },
]
`;
